### PR TITLE
Scan: Binding ledger-cosmos-js

### LINKED ISSUE #1203: Implement binding for ledger-cosmos-js in Scan
Binding [Ledger cosmos js](https://github.com/cosmos/ledger-cosmos-js) on Scan with example code

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #1203 

- [x] Binding `ledger-cosmos-js`
- [x] Send Request tx via ledger
- [x] Implement good abstraction

### CODE REVIEW

File: scan/src/bindings/Ledger.re [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+let getAddress = () => {
+  getApp()
+  |> Js.Promise.then_(app => {
+       //  TODO: remove hard-coded path later
+       let path = [|44, 118, 4, 0, 2|];
+       let prefix = "cosmos";
+       let responsePromise = getAddressAndPubKey(app, path, prefix);
+       let%Promise response = responsePromise;
+       Js.Console.log(response.bech32_address);
```
Comment (sorawit) [inline scan/src/bindings/Ledger.re]: Why do we log?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/wallet/Wallet.re [hunk 2]
```diff
@@ -0,0 +1,24 @@
+type t =
+  | Mnemonic(Mnemonic.t)
+  | Ledger(Ledger.t);
+
+let createFromMnemonic = (rpcUrl, mnemonic) => {
+  let bandChain = BandWeb3.network(rpcUrl, "bandchain");
```
Comment (sorawit) [inline scan/src/wallet/Wallet.re]: I think we can put arbitrary url and chain id here. Reading https://github.com/cosmostation/cosmosjs/blob/master/src/index.js, I don't think it requires url or chain id for getting address or signing.

File: scan/src/wallet/Wallet.re [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  bandChain->BandWeb3.setPath("m/44'/494'/0'/0/0");
+  bandChain->BandWeb3.setBech32MainPrefix("band");
+  Mnemonic({bandChain, mnemonic});
+};
+
+let createFromLedger = path => {
```
Comment (sorawit) [inline scan/src/wallet/Wallet.re]: I think this should be a promise that errors if a ledger is not connected?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Comment by evilpeach** [comment]
@sorawit Currently, there are snake case variables on the binding. Is there a way to handle this properly instead of creating `raw_response_t`(contains snake case) and `response_t` (camel case).

**Review by sorawit — APPROVED** [review]

**Review by tansawit — APPROVED** [review]

**Pull request merged**